### PR TITLE
Feature add options to bool validator

### PR DESCRIPTION
**Pull request by alirezaGAlizade**
**Overview:**
This pull request aims to enhance the boolean validation within Laravel's framework. The existing boolean validator lacked flexibility in checking boolean values with varying degrees of strictness. To address this limitation, this update introduces a more versatile validateBoolean method, allowing for strict, medium, and non-strict validation of boolean values.

**Key Enhancements:**

**Flexibility:** The updated method introduces a parameter for different levels of strictness in boolean validation: 'strict', 'medium', and 'low'.
**Improved Validation:** The method now allows for tailored validation based on the specified strictness level, enabling checks for various boolean representations.

**Why is this PR valuable?**
This improvement provides developers with a more robust and adaptable boolean validation method. It caters to different project requirements by offering multiple levels of validation, ensuring more precise handling of boolean values based on the desired strictness.

This change enhances the framework's flexibility and accommodates a wider range of boolean validation scenarios, contributing to more accurate data validation and handling within Laravel.

### CODE REVIEW

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 1]
```diff
@@ -466,13 +466,20 @@ public function validateBetween($attribute, $value, $parameters)
...
      * @param  mixed  $value
+     * @param  array<int, string>  $parameters
      * @return bool
      */
-    public function validateBoolean($attribute, $value)
+    public function validateBoolean($attribute, $value, $parameters)
     {
-        $acceptable = [true, false, 0, 1, '0', '1'];
+        $strictness = isset($parameters[0]) ? $this->trim($parameters[0]) : 'medium';
+        $defaultAcceptable = [true, false, 0, 1, '0', '1'];
```
Comment (fuwasegu) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php:475]:
```suggestion
        $defaultAcceptable = [true, false, 'true', 'false', 0, 1, '0', '1'];
```

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 2]
```diff
@@ -466,13 +466,20 @@ public function validateBetween($attribute, $value, $parameters)
...
      * @param  string  $attribute
      * @param  mixed  $value
+     * @param  array<int, string>  $parameters
      * @return bool
      */
-    public function validateBoolean($attribute, $value)
+    public function validateBoolean($attribute, $value, $parameters)
     {
-        $acceptable = [true, false, 0, 1, '0', '1'];
+        $strictness = isset($parameters[0]) ? $this->trim($parameters[0]) : 'medium';
```
Comment (fuwasegu) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php:474]:
```suggestion
        $strictness = ($parameters[0] ?? '') === 'strict';
```

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 3]
```diff
@@ -466,13 +466,20 @@ public function validateBetween($attribute, $value, $parameters)
...
+        $defaultAcceptable = [true, false, 0, 1, '0', '1'];
 
-        return in_array($value, $acceptable, true);
+        return match ($strictness) {
+            'strict' => is_bool($value),
+            'medium' => in_array($value, $defaultAcceptable, true),
+            'low' => in_array($value, array_merge($defaultAcceptable, ['true', 'false']), true),
+            default => in_array($value, $defaultAcceptable, true),
+        };
```
Comment (fuwasegu) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php:482]:
```suggestion
        return $strictness
            ? is_bool($value)
            : in_array($value, $defaultAcceptable, true);
```

**Review by fuwasegu — COMMENTED** [review]
I found this refactoring a bit confusing due to the increased number of options.

Since the merging target is master, if we allow destructive changes to be applied from 11.x, we can refactor as follows:

**Comment by taylorotwell** [comment]
You can write your own custom validation rule if you wish.

**Pull request closed without merging**